core/state/statedb.go

```diff
@@ -604,9 +644,30 @@ func (s *StateDB) TryPreload(block *types.Block, signer types.Signer) {
 			accounts[*tx.To()] = true
 		}
 	}
+
+	var overheadCost time.Duration
+	defer func() {
+		goid := cachemetrics.Goid()
+		isSyncMainProcess := cachemetrics.IsSyncMainRoutineID(goid)
+		isMinerMainProcess := cachemetrics.IsMinerMainRoutineID(goid)
+		// record metrics of syncing main process
+		if isSyncMainProcess {
+			syncPreloadCost.Update(overheadCost)
+			syncPreloadCounter.Inc(overheadCost.Nanoseconds())
+		}
+		// record metrics of mining main process
+		if isMinerMainProcess {
+			minerPreloadCost.Update(overheadCost)
+			minerPreloadCounter.Inc(overheadCost.Nanoseconds())
+		}
+	}()
+
+	start := time.Now()
 	for account := range accounts {
 		accountsSlice = append(accountsSlice, account)
 	}
+	overheadCost = time.Since(start)
```

Comment:
Do you think we should move `	overheadCost = time.Since(start)` down?

Reply:
yes ,  it  should move down

Reply:
updated